update triangle tests

DebbieSan opened the pull request:
Hello,

I followed the guide to synch and update the tests for the triangle exercise. I am not sure if there is another part to this process. If yes, please advise.

Thank you kindly!

IsaacG commented on exercises/practice/triangle/.meta/tests.toml:
```diff
@@ -1,60 +1,73 @@
-description = "two sides are equal"
+description = "scalene triangle -> first and second sides are equal"
+
+[3da23a91-a166-419a-9abf-baf4868fd985]
+description = "scalene triangle -> first and third sides are equal"
+
+[b6a75d98-1fef-4c42-8e9a-9db854ba0a4d]
+description = "scalene triangle -> second and third sides are equal"
```
This says there are additional tests being added. Did you add those tests to the unit test file?

DebbieSan replied:
I might not have done so, I ran the confliglet commands and checked the files to ensure nothings was changed in the config file. Would that be the implementation step?

IsaacG replied:
I'm not sure what set of steps you're following to comment on what it would be labeled. 

These UUIDs map to test cases in [the canonical data](https://github.com/exercism/problem-specifications/blob/main/exercises/triangle/canonical-data.json) and need to line up with [the C++ test cases](https://github.com/exercism/cpp/blob/main/exercises/practice/triangle/triangle_test.cpp). The ask isn't to run a command. The ask is to update the test files -- both the toml metadata and the unit tests. They need to be done at the same time and be kept in sync.

DebbieSan replied:
Hi, I apologize as my answer wasn't clear. I did try to update the tests. I will rectify this and update my PR. Thank you.

vaeng replied:
Are you still working on this @DebbieSan? Do you need help?

DebbieSan replied:
Hi @vaeng, I did get caught up with school but I am working on this. I will pick it back up today. I have a question though about the tests. 

```
// canonical-data.json tests this with a scalene triangle?
TEST_CASE("first_and_third_sides_are_equal")
{
    REQUIRE(triangle::flavor::isosceles == triangle::kind(3, 4, 3));
}

// canonical-data.json tests this with a scalene triangle?
TEST_CASE("second_and_third_sides_are_equal")
{
    REQUIRE(triangle::flavor::isosceles == triangle::kind(4, 3, 3));
}
```

The triangles in the test case have property:scalene but they are isosceles. Is this correct? Are the tests just suppose to give a true/false state? 

Thank you for your assistance.

vaeng replied:
Yes, the REQUIRE Test macro expects a true or false inside the brackets, so most tests use a `==` comparison between an expected value and the one the student-implemented function returns.

vaeng replied:
@DebbieSan 
Are you good to move on, or do you have more questions?

DebbieSan replied:
Hello, I updated the PR.

vaeng commented:
Would you be so kind and add the uuids from the toml file to the tests and change the names of the cpp test files so they match the toml file as well?
Here is an example:
https://github.com/exercism/cpp/blob/main/exercises/practice/allergies/allergies_test.cpp

I can do it, if you don't want to spend more time on this. I know it can be exhausting to change so many things one by one.

DebbieSan commented:
hi @vaeng I will do it :)

DebbieSan commented:
Hi @vaeng, I added the UUIDs and updated the names, a couple of the tests I was unsure as they did not seem to match the toml file. Please advise and thank you kindly for your assistance.

vaeng commented:
Hello @DebbieSan,

please excuse my commit, it is a complete rework of your submission. I hope you are okay with this.
There were a bunch of smaller issues (forgotten brackets, quotes, doubled UUIDS, and some other smaller things), and pointing out all them would be more work for both of us than just correcting them.

Also, I could not get the actions to trigger a check, so I used a commit to run them. Now we can see the check.

A bigger issue was that the tests as they are specified in the problem spec, are not completely compatible with the way the exercise solution is set up on this track. I had to mark them disabled and removedthem from the test file.

I hope you are okay with this. It took a long time for me to come back to you, but you will still get your exercism reputation points. Better late than never. I also threw in the hacktoberfest-contribution so you have a small jump-start from the past to continue your open-source progress.

Thanks for all the work and your patience.

vaeng reviewed (approved):
[no important files changed]
